src/main/java/mate/academy/bookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.bookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import mate.academy.bookstore.dto.user.UserRegistrationRequestDto;
+
+public class FieldMatchValidator implements
+        ConstraintValidator<FieldMatch, UserRegistrationRequestDto> {
+    private String firstFieldName;
+    private String secondFieldName;
+    private String message;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+        this.message = constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(UserRegistrationRequestDto value, ConstraintValidatorContext context) {
+        if (value == null) {
```

Comment:
It's better to simplify 

See example
```suggestion
        Object field = new BeanWrapperImpl(value).getPropertyValue(this.field);
        Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.fieldMatch);
        return Objects.equals(field, fieldMatch);
```